src/libs/ReportUtils.ts

```diff
@@ -1565,6 +1565,17 @@ function isOneTransactionReport(reportID: string): boolean {
     return ReportActionsUtils.getOneTransactionThreadReportID(reportID, reportActions) !== null;
 }
 
+/*
+ * Whether the report contains only one expense and the expense should be paid later
+ */
+function isPayAtEndExpenseReport(reportID: string, transactions: Transaction[] | undefined): boolean {
+    if ((!!transactions && transactions.length !== 1) || !isOneTransactionReport(reportID)) {
+        return false;
+    }
+
+    return TransactionUtils.isPayAtEndExpense(transactions?.[0] ?? TransactionUtils.getAllReportTransactions(reportID)[0]);
```

Comment:
```suggestion
    return TransactionUtils.isPayAtEndExpense(transactions[0]);
```

`transaction[0]` should definitely be defined here based on the checks above.

Reply:
Oh, actually...not necessarily because transactions is optional because when calling from `approveRequests` we don't have the transactions.

Reply:
Ah okay I read the conditions incorrectly